app/services/user_audit_logger.rb

```diff
@@ -35,19 +33,30 @@ def validate_required_fields
   end
 
   def validate_user_action_event
-    raise MissingUserActionEventError, 'User action event must be present' if user_action_event.nil?
+    raise 'User action event must be present' if user_action_event.nil?
   end
 
   def validate_subject_verification
-    raise MissingSubjectVerificationError, 'Subject user verification must be present' if subject_user_verification.nil?
+    raise 'Subject user verification must be present' if subject_user_verification.nil?
```

Comment:
Should we just use the rails class validators for these?

Reply:
and then you can replace `validate_required_fields` with `validate!`

Reply:
Right, the  UserAction is covered already by `UserAction.create!` & I guess UserActionEvent is covered by `UserActionEvent.find_by` since it won't find one with a bad event  identifier and the events coming from the DB will be valid.

So I'll just run `subject_user_verification.validate!`.